src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs

```diff
@@ -288,15 +289,15 @@ internal static TempAssembly GenerateTempAssembly(XmlMapping xmlMapping, Type ty
                 return null;
             }
 
-            return new TempAssembly(new XmlMapping[] { xmlMapping }, new Type[] { type }, defaultNamespace, location);
+            return new TempAssembly(new XmlMapping[] { xmlMapping }, new Type?[] { type }, defaultNamespace, location);
         }
 
         public void Serialize(TextWriter textWriter, object o)
         {
             Serialize(textWriter, o, null);
         }
 
-        public void Serialize(TextWriter textWriter, object o, XmlSerializerNamespaces namespaces)
+        public void Serialize(TextWriter textWriter, object o, XmlSerializerNamespaces? namespaces)
```

Comment:
```suggestion
        public void Serialize(TextWriter textWriter, object? o, XmlSerializerNamespaces? namespaces)
```
I'm pretty sure you should be able to serialize `null`.
I did a sample app and found no problem doing so.

Reply:
I agree, not sure how I missed that. There is one NRE scenario there (which could have chain lead to this) if user passed XmlMembersMapping to a constructor but seems it only has internal constructor so `!` that callsite